privacyidea/lib/resolvers/LDAPIdResolver.py

```diff
@@ -493,57 +496,71 @@ def _getDN(self, userId):
 
         :param userId: The userid of a user
         :type userId: string
-
         :return: The DN of the object.
         """
         dn = ""
         if self.uidtype.lower() == "dn":
             dn = userId
         else:
             # get the DN for the Object
-            self._bind()
-            search_userId = self._trim_user_id(userId)
-            filter = "(&{0!s}({1!s}={2!s}))".format(self.searchfilter,
-                                                    self.uidtype,
-                                                    search_userId)
-            self.l.search(search_base=self.basedn,
-                          search_scope=self.scope,
-                          search_filter=filter,
-                          attributes=list(self.userinfo.values()))
-            r = self.l.response
-            r = self._trim_result(r)
+            search_userid = self._trim_user_id(userId)
+            search_filter = f"(&{self.searchfilter}({self.uidtype}={search_userid}))"
+            r = self._search(search_base=self.basedn, search_filter=search_filter,
+                             attributes=list(self.userinfo.values()))
             if len(r) > 1:  # pragma: no cover
-                raise Exception("Found more than one object for uid {0!r}".format(userId))
+                raise ResolverError(f"Found more than one object for uid {userId!r}")
             elif len(r) == 1:
                 dn = r[0].get("dn")
             else:
-                log.info("The filter {0!r} returned no DN.".format(filter))
-
+                log.info("The filter {0!r} returned no DN.".format(search_filter))
         return dn
 
     def _bind(self):
+        """
+        Perform LDAP bind operation on a connection.
+        Create the connection if it doesn't exist yet
+        """
         if not self.i_am_bound:
             if not self.serverpool:
                 self.serverpool = self.get_serverpool_instance(self.get_info)
-            self.l = self.create_connection(authtype=self.authtype,
-                                            server=self.serverpool,
-                                            user=self.binddn,
-                                            password=self.bindpw,
-                                            receive_timeout=self.timeout,
-                                            auto_referrals=not
-                                            self.noreferrals,
-                                            start_tls=self.start_tls,
-                                            keytabfile=self.keytabfile)
-            if not self.l.bind():
-                raise Exception("Wrong credentials")
+            try:
+                self.l = self.create_connection(authtype=self.authtype,
```

Comment:
does it have to be called `l`? Could be something more expressive, if its created by `create_connection` something like `connection`, `ldap_connection`